File: java/quarkus/1 - quarkus-basics/src/main/java/com/r2m/infrastructure/persistence/repository/PackageRepository.java
package com.r2m.infrastructure.persistence.repository;

import com.r2m.infrastructure.persistence.entity.PackageEntity;
import io.quarkus.hibernate.reactive.panache.PanacheRepository;
import jakarta.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class PackageRepository implements PanacheRepository<PackageEntity> {
}

File: kotlin/2-package-manager-api/package-manager-api/src/main/kotlin/com/r2m/package_manager_api/infrastructure/persistence/repositories/PackageRepository.kt
package com.r2m.package_manager_api.infrastructure.persistence.repositories

import com.linecorp.kotlinjdsl.support.spring.data.jpa.repository.KotlinJdslJpqlExecutor
import com.r2m.package_manager_api.infrastructure.persistence.entity.PackageEntity
import com.r2m.package_manager_api.infrastructure.persistence.repositories.impl.CustomPackageRepository
import org.springframework.data.repository.ListCrudRepository

interface PackageRepository : ListCrudRepository<PackageEntity, String>, CustomPackageRepository, KotlinJdslJpqlExecutor {
}